### PR TITLE
Adding suggestion reconcile in Experiment controller

### PR DESCRIPTION
**Pull request by johnugeorge**
This initial PR tracks the reconcile logic of suggestions in experiment controller. 

Minor changes are included in Suggestion CR. Name in trial assignment is populated by Suggestion Controller when suggestions are populated.

Suggestion controller is still using fake suggestion interface to generate suggestions. It will be fixed in follow up PRs. 

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/750)

### CODE REVIEW

File: pkg/controller.v1alpha3/experiment/suggestion/suggestion.go [hunk 1]
```diff
@@ -1,31 +1,83 @@
...
-	if reply, err := common.GetSuggestions(request); err != nil {
+func (g *General) GetOrCreateSuggestion(instance *experimentsv1alpha3.Experiment, suggestionRequests int32) (*suggestionsv1alpha3.Suggestion, error) {
+	logger := log.WithValues("experiment", types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace})
+	suggestion := &suggestionsv1alpha3.Suggestion{}
+	err := g.Get(context.TODO(),
+		types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}, suggestion)
+	if err != nil && errors.IsNotFound(err) {
+		logger.Info("Creating Suggestion", "namespace", instance.Namespace, "name", instance.Name, "requests", suggestionRequests)
+		if err := g.createSuggestion(instance, suggestionRequests); err != nil {
+			return nil, err
```
Comment (gaocegege) [inline pkg/controller.v1alpha3/experiment/suggestion/suggestion.go]: LGTM while I think we should have a comment to show why we should return an error here.

Comment (johnugeorge) [reply pkg/controller.v1alpha3/experiment/suggestion/suggestion.go]: Added logs

**Comment by johnugeorge** [comment]
@gaocegege @hougangliu 

Are we ready to merge this?

**Comment by gaocegege** [comment]
/hold

Wait for @hougangliu 's approval

**Comment by hougangliu** [comment]
> /hold
> 
> Wait for @hougangliu 's approval

Thanks! I am reviewing it now

**Pull request merged**